### PR TITLE
Adapt simumlation physics to take into account qualesi models

### PR DESCRIPTION
**Pull request by axrolld**
WIP !

### CODE REVIEW

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java [hunk 1]
```diff
@@ -60,9 +41,11 @@ static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCur
...
     double getMaxBrakingForce(double speed);
 
-    /** The maximum acceleration, in m/s^2, which can be applied at a given speed, in m/s */
-    record TractiveEffortPoint(double speed, double maxEffort) {
-    }
+    /** Returns the different energy sources of the rolling stock */
+    ArrayList<EnergySource> getEnergySources();
+
+    /** If relevant, update the different energy storages states of charge */
+    void updateEnergyStateOfCharge(double force, double speed, double timeStep);
```
Comment (lmuria) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java:48]: would find it easier to read with a "Storage" somwhere in the name like :
updateEnergySourceStorageStateOfCharge || updateESStorageStateOfCharge || updateEnergyStorageStateOfCharge

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/power/Battery.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                pMin,
+                pMax,
+                new EnergyStorage(
+                        capacity,
+                        0.8,
+                        0.2, 0.9,
+                        new RefillLaw(97.6,0.8, capacity),
+                        new SocDependantPowerCoefficient(curveHigherValueOnHighSpeed)
+                ),
+                1
```
Comment (lmuria) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/power/Battery.java]: Shouldn't 1 be 1.0 here ?

Comment (axrolld) [reply core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/power/Battery.java]: Yes you're right, i'll change this. But those values will have to change anyways

File: core/src/main/java/fr/sncf/osrd/train/RollingStock.java [hunk 3]
```diff
@@ -130,21 +143,64 @@ public double getRollingResistanceDeriv(double speed) {
...
+        return availablePower;
+    }
+
     @Override
     public double getMaxBrakingForce(double speed) {
         return gamma * inertia;
     }
 
+    @Override
+    public void updateEnergyStorages(double maxAvailableForce, double usedForce, double speed, double timeStep) {
```
Comment (lmuria) [inline core/src/main/java/fr/sncf/osrd/train/RollingStock.java:177]: TBC

**Comment by axrolld** [comment]
😿

**Pull request closed without merging**